platform/mds/mds/src/main/java/org/motechproject/mds/service/DefaultMotechDataService.java

```diff
@@ -317,8 +317,11 @@ public T revertFromTrash(Long trashId) {
             T newInstance = getClassType().newInstance();
 
             copyValuesFromRecord(newInstance, trashRecord);
+            newInstance = create(newInstance);
 
-            return create(newInstance);
+            trashService.moveFromTrash(newInstance, trashRecord, recordHistory);
```

Comment:
I've taken a look at the implementation of this method and it makes no sense. Can you make the following changes:

* remove the newInstance and recordHistory params, since they are unused
* rename the method to removeFromTrash